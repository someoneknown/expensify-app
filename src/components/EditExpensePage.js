import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        console.log('Expense Object is', this.props.expense);
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onClick = (e) => {
        this.props.startRemoveExpense(this.props.expense);
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <ExpenseForm 
                expense = {this.props.expense}
                onSubmit = {this.onSubmit}
                />
                <button
                onClick = { this.onClick }
                >Remove
                </button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startRemoveExpense: id => {
        dispatch(startRemoveExpense(id));
    },
    startEditExpense: (id, updates) => {
        dispatch(startEditExpense(id, updates));
    }
});

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);