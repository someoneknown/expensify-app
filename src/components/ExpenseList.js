import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectedExpenses from '../selectors/expenses';

export  const ExpenseList  = (props) => (
    <div>
        {props.expenses.length ? (
            props.expenses.map((expense) =>
            <ExpenseListItem 
                {...expense}
                key = {expense.id} 
            />
        )
        ) : (
            <p>No Expenses</p>
        )}
    </div>
);
const mapStatetoProps = (state) => {
    return {
        expenses: selectedExpenses(state.expenses, state.filters)
    };
};
export default connect(mapStatetoProps)(ExpenseList);
 