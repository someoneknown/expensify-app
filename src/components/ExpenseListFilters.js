import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (focused) => {
        this.setState(() => ({ calendarFocused: focused}));
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount();
    }
    render() {
        return (
            <div>
                <input
                    type = "text"
                    value = {this.props.filters.text}
                    onChange = {this.onTextChange}/>
                <select
                    value={this.props.filters.sortBy} onChange={this.onSortChange}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={() => false}
                showClearDates
                displayFormat={'DD/MM/YYYY'}
                />
            </div>
        );
    }
}   

const mapStateToProps = (state) => (
    {
        filters: state.filters
    }
);

const mapDispatchToProps = dispatch => ({
    setTextFilter: value => {
        dispatch(setTextFilter(value));
    },
    sortByDate: _ => {
        dispatch(sortByDate());
    },
    sortByAmount: _ => {
        dispatch(sortByAmount());
    },
    setStartDate: startDate => {
        dispatch(setStartDate(startDate));
    },
    setEndDate: endDate => {
        dispatch(setEndDate(endDate ));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);