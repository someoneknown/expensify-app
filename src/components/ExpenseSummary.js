import React from 'react';
import numeral from 'numeral';
import getVisibleExpenese from '../selectors/expenses';
import getTotalExpense from '../selectors/expense-total';
import { connect } from 'react-redux';

export class ExpenseSummary extends React.Component {

    render() {
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {this.props.expenseCount === 1 ? 'expense' : 'expenses'} totatling {numeral(this.props.expenseTotal / 100).format('$0,00.00')}</p>
            </div>
        );
    }
};

const mapStatetoProps = ({ expenses, filters }) => {
    const visibleExpenese = getVisibleExpenese(expenses, filters);
    const expenseCount = visibleExpenese.length;
    const expenseToatal = getTotalExpense(visibleExpenese);
    return {
        expenseCount,
        expenseTotal
    };
};

export default connect(mapStatetoProps)(ExpenseSummary);