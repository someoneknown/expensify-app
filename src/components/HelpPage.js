import React from 'react';

const HelpPage = _ => (
    <div>
        This is my help component.
    </div>
);

export default HelpPage;