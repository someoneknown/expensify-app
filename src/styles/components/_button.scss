.button {
    background: $blue;
    color: #fff;
    border: none;
    display: inline-block;
    font-weight: 400; 
    font-size: $font-size-large;
    padding: $s-size;
    line-height: 1;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}