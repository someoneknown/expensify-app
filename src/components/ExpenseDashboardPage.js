import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = _ => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;