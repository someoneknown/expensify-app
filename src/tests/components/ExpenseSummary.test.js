import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme'; 
import React from 'react';

test('should render ExpenseSummary with single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={120}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={2} expenseTotal={12030.0}/>);
    expect(wrapper).toMatchSnapshot();
});
