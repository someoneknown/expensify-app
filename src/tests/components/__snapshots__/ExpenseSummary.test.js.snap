// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totatling 
    $120.30
  </p>
</div>
`;

exports[`should render ExpenseSummary with single expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totatling 
    $1.20
  </p>
</div>
`;
